### PR TITLE
Feature thomas

### PR DESCRIPTION
**Pull request by ttridon**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,14 +5,18 @@
   "dependencies": {
     "@blueprintjs/core": "^1.35.0",
     "classnames": "^2.2.5",
+    "deep-freeze": "0.0.1",
+    "deepfreeze": "^2.0.0",
```
Comment (nicolas-briemant) [inline package.json]: why 2 deps ?

File: package.json [hunk 2]
```diff
@@ -5,14 +5,18 @@
...
     "faker": "^4.1.0",
     "glamor": "^2.20.40",
     "glamorous": "^4.11.2",
     "prop-types": "^15.6.0",
     "react": "^16.2.0",
     "react-addons-css-transition-group": "^15.6.2",
     "react-dom": "^16.2.0",
+    "react-redux": "^5.0.6",
```
Comment (nicolas-briemant) [inline package.json:16]: no redux in this PR

File: src/App.css [hunk 3]
```diff
@@ -12,3 +12,7 @@
 .notDisplay {
   display: none;
 }
+
```
Comment (nicolas-briemant) [inline src/App.css]: no new style in CSS

File: src/App.js [hunk 4]
```diff
@@ -3,16 +3,25 @@ import PropTypes from 'prop-types';
 import './App.css';
 import Header from './components/missions/header';
 import Missions from './components/missions/sections';
+import * as create from './actions';
```
Comment (nicolas-briemant) [inline src/App.js]: bad name: create

File: src/components/logger.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable */
```
Comment (nicolas-briemant) [inline src/components/logger.js]: too wide, use disable line only

File: src/components/missions/sections/index.js [hunk 6]
```diff
@@ -1,33 +1,74 @@
...
-  <CellMission interactive="true" elevation={Card.ELEVATION_TWO}>
-    <b>{name}</b>
-    <p>
-      clientId: {clientId}, partnerId: {partnerId}, managerId: {managerId}
-    </p>
-    <ul>{addenda.map(worker => <Worker key={uniqid()} {...worker} />)}</ul>
-  </CellMission>
-);
+const DisplayButton = glamorous.div({}, ({ isHovered }) => ({
+  display: isHovered ? 'block' : 'none',
```
Comment (nicolas-briemant) [inline src/components/missions/sections/index.js]: don't always render a Component and play with style to see it or not,
render it only when you want to see it.

File: src/reducers/index.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  return missions.map(mission => {
+    return mission.id === id ? { ...mission, isSelected: !mission.isSelected } : mission;
+  });
+};
+
+const findIndexMission = (missions, id) => missions.findIndex(mission => mission.id === id);
+
+const remove = (missions, id) => {
+  const idByIndex = findIndexMission(missions, id);
+  return missions.filter((mission, index) => idByIndex !== index);
```
Comment (nicolas-briemant) [inline src/reducers/index.js]: use only filter

**Review by nicolas-briemant — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ttridon) [reply package.json]: I installed the wrong one.

File: package.json (hunk 2 above)
Comment (ttridon) [reply package.json:16]: I didn't use it, just installed.

File: src/App.js [hunk 8]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 import './App.css';
 import Header from './components/missions/header';
 import Missions from './components/missions/sections';
-import * as create from './actions';
+import * as createAction from './actions';
```
Comment (nicolas-briemant) [inline src/App.js]: le mieux c'est actionCreators car il s'agit d'un obj et non d'une function à exécuter

File: src/components/logger.js [hunk 9]
```diff
@@ -1,12 +1,11 @@
-/* eslint-disable */
-
 import React from 'react';
 import PropTypes from 'prop-types';
 
 export default id => Component => {
+  // eslint-disable-next-line
   class Logger extends React.Component {
```
Comment (nicolas-briemant) [inline src/components/logger.js]: il ne te manque pas le stateless disabling ?

File: src/components/missions/sections/index.js [hunk 10]
```diff
@@ -58,10 +52,12 @@ class Mission extends Component {
           clientId: {clientId}, partnerId: {partnerId}, managerId: {managerId}
         </p>
         <ul>{addenda.map(worker => <Worker key={uniqid()} {...worker} />)}</ul>
-        <DisplayButton isHovered={this.state.isHovered}>
-          <Button iconName="select" text="select" onClick={() => selectMission(id)} />
-          <Button iconName="trash" text="remove" onClick={() => removeMission(id)} />
-        </DisplayButton>
+        {this.state.isHovered ? (
+          <div isHovered={this.state.isHovered}>
```
Comment (nicolas-briemant) [inline src/components/missions/sections/index.js]: la prop isHovered n'est plus utile

**Review by nicolas-briemant — CHANGES_REQUESTED** [review]
almost perfect !

**Review by nicolas-briemant — APPROVED** [review]

**Pull request merged**